server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1763,18 +1763,26 @@ public static void toXContent(IndexMetadata indexMetadata, XContentBuilder build
                 }
                 builder.endArray();
             } else {
-                builder.startObject(KEY_MAPPINGS);
-                MappingMetadata mmd = indexMetadata.mapping();
-                if (mmd != null) {
-                    Map<String, Object> mapping = XContentHelper.convertToMap(mmd.source().uncompressed(), false).v2();
-                    if (mapping.size() == 1 && mapping.containsKey(mmd.type())) {
-                        // the type name is the root value, reduce it
-                        mapping = (Map<String, Object>) mapping.get(mmd.type());
+                final MappingMetadata mmd = indexMetadata.mapping();
+                if (params.paramAsBoolean(Metadata.MAPPINGS_BY_HASH_PARAM, true)) {
+                    if (mmd == null) {
+                        builder.nullField(KEY_MAPPINGS);
+                    } else {
+                        builder.field(KEY_MAPPINGS, indexMetadata.mapping().getSha256());
```

Comment:
I would prefer to pick a new name for the SHA256. `mappings_sha256` maybe or perhaps `shared_mappings_id`. Then we can also allow dumping both, which could provide a transitional path for clients.